base/replutil.jl

```diff
@@ -147,11 +147,13 @@ function showerror(io::IO, ex::BoundsError)
         print(io, ": attempt to access ")
         if isa(ex.a, AbstractArray)
             print(io, summary(ex.a))
+        elseif isa(ex.a, Tuple) && isdefined(ex, :i) && isa(ex.i, Tuple)
```

Comment:
Option 4: don't support the throwing version of `checkbounds_indices`. Really I was most concerned about setting up an expectations that `checkbounds_indices(IA, I)` would throw, and so if that yields a `MethodError` but `checkbounds_indices(Bool, IA, I)` works, any user will expect that to return a `Bool` and then could handle the throwing him/herself. Maybe this is the best choice?